module4/src/main/java/com/remiz/module4/Main.java

```diff
@@ -0,0 +1,207 @@
+                Fridge.class, Heater.class, Microwave.class, Multicooker.class, PS.class, Stove.class, TV.class, VacuumCleaner.class,
+                Washer.class, Devices.class);
+        Marshaller marshaller = context.createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(devices, new File("./devices.xml"));
+
+//        JAXBContext context = JAXBContext.newInstance(ElectricDevice.class, AirConditioner.class, Computer.class, DishWasher.class,
+//                Fridge.class, Heater.class, Microwave.class, Multicooker.class, PS.class, Stove.class, TV.class, VacuumCleaner.class,
+//                Washer.class, Devices.class);
+//        Unmarshaller unmarshaller = context.createUnmarshaller();
+//        Devices unmarshalledDevices = (Devices) unmarshaller.unmarshal(new FileReader("./devices.xml"));
+//        System.out.println(unmarshalledDevices.toString());
+//
+//        List<ElectricDevice> devicesList = unmarshalledDevices.getDeviceList();
+
+        ScannerWrapper scanner = new ScannerWrapper();
+        int check = 0;
+        try {
+            do {
+                scanner.openScanner();
+                System.out.println("-1.Switched on devise\n-2.Calculate the power sum consumption" +
+                        "\n-3.Sort devices by power\n-4.Find the required device by range\n-5.What devices can do?" +
+                        "\n-6.Finish work");
+                int enteredNumber = scanner.readInput();
+                if (enteredNumber == 1) {
+                    switchOnDevices(devicesList);
+                } else if (enteredNumber == 2) {
+                    calculateSumPower(devicesList);
+                } else if (enteredNumber == 3) {
+                    sortElectricDevicesByPower(devicesList);
+                } else if (enteredNumber == 4) {
+                    do {
+                        System.out.println("Choose a suitable power range:\n-1.150-450\n-2.900-1000\n-3.1000-1500\n" +
+                                "-4.Go to main menu");
+                        check = scanner.readInput();
+                        findDeviceByRange(check, devicesList);
+                    } while (check != 4);
+                } else if (enteredNumber == 5) {
+                    do {
+                        System.out.println("What doing:\n-1.cleaning devices\n-2.cooking devices\n-3.entertaining devices" +
+                                "\n-4.changing temperature devices\n-5.Go to main menu");
+                        check = scanner.readInput();
+                        outputDeviceAction(check, devicesList);
+                    } while (check != 5);
+                } else if (enteredNumber == 6) {
+                    System.out.println("Bye-bye!");
+                    check = 1;
+                } else {
+                    System.out.println("Please,enter number from 1 to 5");
+                }
+            } while (check != 1);
```

Comment:
Please move `while` at the top of the cycle.
You don't need `do` section in this case.